be/src/exec/vectorized/aggregate/aggregate_blocking_node.cpp

```diff
@@ -27,6 +28,11 @@ Status AggregateBlockingNode::open(RuntimeState* state) {
 
     VLOG_ROW << "group_by_expr_ctxs size " << _aggregator->group_by_expr_ctxs().size() << " _needs_finalize "
              << _aggregator->needs_finalize();
+    bool agg_group_by_with_limit =
+            (!_aggregator->is_none_group_by_exprs() &&     // has group by
+             _limit != -1 &&                               // has limit
+             _conjunct_ctxs.empty() &&                     // no 'having' clause
+             _aggregator->get_aggr_phase() == AggrPhase2); // phase 2, keep it to make things safe
```

Comment:
It's only because of phase2 or because we need to make sure we run it on the most top level? 

For example, if there are 3-stages, should we still stick to phase2 or phase3?

Reply:
In BE, aggregate only have Phase1 or Phase2, no phase3. For simply, you could think Phase2 means blocking, Phase1 means streaming.  Maybe we could remove Phase1, Phase2 conception in BE